rotkehlchen/rotkehlchen.py

```diff
@@ -336,6 +333,7 @@ def unlock_user(
 
         with self.data.db.conn.read_ctx() as cursor:
             settings = self.get_settings(cursor)
+            CachedSettings().update_entries(settings)  # update cached settings from DB
```

Comment:
oooof so it was never initialized? Damn ...

So the main thing here is the logic needs changing.

`update_entries` should only get a `ModifiableDBSettings` argument. So here you need to turn the returned settings to a ModifiableDBSettings or just more simply create an `initialize()` function which just does what `reset()` does but with an argument of `DBSettings`.